src/gcl/MeshVelocityAlg.C

```diff
@@ -135,7 +135,12 @@ MeshVelocityAlg<AlgTraits>::execute()
         }
       }
 
-      for (int ip = 0; ip < AlgTraits::numScsIp_; ++ip) {
+      int nip = std::min(12,AlgTraits::numScsIp_);
+      if (AlgTraits::numScsIp_ > nip) {
+        std::cerr << "Error: MeshVelocityAlg can't support more than 12 integration points." << std::endl;
+      }
+
+      for (int ip = 0; ip < nip; ++ip) {
```

Comment:
@alanw0 This algorithm is only available for Hex8, Line 57 has a throw to disallow any other element types. The `std::cerr` message is not necessary.

Reply:
Hey Shreyas!
Ah, ok, I hadn't noticed line 57. I can remove the cerr. Of course it won't get hit anyway, since line 57 will stop things before we get that far. But adding the std::min line was necessary to get the compiler to stop noticing that the Hex27 template instantiation could execute the following for-loop more than 12 times which would over-run array bounds.

Reply:
@sayerhs Yes I'll do that. And also use AlgTraitsHex8::numScsIp_ in the declaration of the array at line 69 of MeshVelocityAlg.h.